volumes:
  postgres-data:
    external: false # Внешние волюмы не привязываются к композу и при docker-compose down --volumes не удаляются
  redis-data:
    external: false

networks:
  litminka-net: 
    external: false

services:
  postgre-db:
    container_name: litminka-db
    image: postgres:15-alpine
    environment:
      POSTGRES_DB: "litminka"
      POSTGRES_USER: "litminka"
      POSTGRES_PASSWORD: "password"
    ports:
      - target: 5432
        published: 5432
    volumes:
      - postgres-data:/var/lib/postgresql/data
    networks:
      - litminka-net

  redis:
    container_name: redis
    image: redis:7-alpine
    volumes:
      - redis-data:/data
    ports:
      - target: 6379
        published: 6379
    networks:
      - litminka-net

  nginx-proxy:
    container_name: nginx-proxy
    image: nginx:alpine
    ports:
      - target: 80
        published: 80
      - target: 443
        published: 443
    volumes:
      - ./cert:/etc/nginx/ssl:ro
      - ./nginx.docker.conf:/etc/nginx/nginx.conf:ro
    networks:
      - litminka-net

  litminka-api:
    image: litminka-api:latest
    container_name: litminka-api
    build:
      context: .
      dockerfile: Dockerfile
      tags:
        - "litminka-api:v2.0.0-alpha.1"
        - "litminka-api:alpine"
    env_file: .env.docker
    networks:
      - litminka-net
